pkg/ddl/index.go

```diff
@@ -346,17 +347,20 @@ func BuildIndexInfo(
 		Unique:  isUnique,
 	}
 
-	if isVector {
+	switch columnarIndexType {
+	case model.ColumnarIndexTypeVector:
 		vectorInfo, _, err := buildVectorInfoWithCheck(indexPartSpecifications, tblInfo)
```

Comment:
Can we move this to somewhere common (maybe an interface?)? In case of we forget to add some other variants some day

Reply:
I failed to find a good way to achieve this...